alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/GlobalConfigExistsValidator.java

```diff
@@ -42,23 +44,33 @@
     public static final String GLOBAL_CONFIG_MISSING = "%s global configuration missing.";
     private static final Logger logger = LoggerFactory.getLogger(GlobalConfigExistsValidator.class);
     private ConfigurationAccessor configurationAccessor;
-    private List<DescriptorKey> descriptorKeys;
+    private List<Descriptor> descriptors;
 
     @Autowired
-    public GlobalConfigExistsValidator(ConfigurationAccessor configurationAccessor, List<DescriptorKey> descriptorKeys) {
+    public GlobalConfigExistsValidator(ConfigurationAccessor configurationAccessor, List<Descriptor> descriptors) {
         this.configurationAccessor = configurationAccessor;
-        this.descriptorKeys = descriptorKeys;
+        this.descriptors = descriptors;
     }
 
+    /**
+     * @return An Optional<String> containing the error message.
+     */
     public Optional<String> validate(String descriptorName) {
         if (StringUtils.isBlank(descriptorName)) {
             return Optional.empty();
         }
-        String descriptorDisplayName = descriptorKeys.stream()
-                                           .filter(descriptorKey -> descriptorKey.getUniversalKey().equals(descriptorName))
-                                           .map(descriptorKey -> descriptorKey.getDisplayName())
-                                           .findFirst()
-                                           .orElse(descriptorName);
+
+        Optional<DescriptorKey> optionalDescriptorKey = descriptors
+                                                            .stream()
+                                                            .filter(desc -> desc.getDescriptorKey().getUniversalKey().equals(descriptorName))
+                                                            .filter(this::hasGlobalConfig)
```

Comment:
Validating global configs is more expensive than comparing the descriptorName which is why the map happens later.